source/store/actions/dynamicFormData.ts

```diff
@@ -0,0 +1,103 @@
+import { Form } from '../../types/FormTypes';
+import { Case } from '../../types/CaseType';
+import { User } from '../../types/UserTypes';
+import { convertAnswerArrayToObject } from '../../helpers/DataStructure';
+
+const addIds = (descriptor: string[], questionId: string, formId: string) =>
+  descriptor.map(s => {
+    if (s.split('.')[0] !== 'user' && questionId !== '') {
+      return `${formId}.${questionId}.${s}`;
+    } else if (s.split('.')[0] !== 'user' && questionId === '') {
+      return `${formId}.${s}`;
+    }
+    return s;
+  });
+
+/** Takes a form and generates the dataMap template used to fill the dynamical data.  */
+const makeDataMap = (form: Form) => {
+  const dataMap = {};
+  form.steps.forEach(step => {
+    if (!step.questions) {
+      return;
+    }
+    step.questions.forEach(qs => {
+      if (qs.loadPrevious) {
+        dataMap[qs.id] = addIds(qs.loadPrevious, '', form.id);
+      } else if (qs.type === 'substepList' && qs.items) {
+        const items = {};
+        qs.items.forEach(item => {
+          if (item.loadPrevious) {
+            items[item.title] = addIds(item.loadPrevious, qs.id, form.id);
+          }
+        });
+        if (Object.keys(items).length > 0) {
+          dataMap[qs.id] = items;
+        }
+      } else if (qs.type === 'editableList' && qs.inputs) {
+        const inputs = {};
+        qs.inputs.forEach(input => {
+          if (input.loadPrevious) {
+            inputs[input.key] = addIds(input.loadPrevious, qs.id, form.id);
+          }
+        });
+        if (Object.keys(inputs).length > 0) {
+          dataMap[qs.id] = inputs;
+        }
+      }
+    });
+  });
+  return dataMap;
+};
+
+const sortCasesByLastUpdated = (list: Case[]) => list.sort((a, b) => b.updatedAt - a.updatedAt);
+
+const treeParseAcc = (obj: Record<string, any> | string[], parser: (s: string[]) => any) => {
+  if (Array.isArray(obj) && obj.every(s => typeof s === 'string')) {
+    const parseRes = parser(obj);
+    return parseRes === '' ? undefined : parseRes;
+  }
+  const res: Record<string, any> = {};
+  Object.keys(obj).forEach(key => {
+    res[key] = treeParseAcc(obj[key], parser);
+  });
+  return res;
+};
+
+const treeParse = (obj: Record<string, any>, parser: (s: string[]) => any): Record<string, any> => treeParseAcc(obj, parser);
```

Comment:
Well, now the treeParseAcc function can be renamed treeParse, and the treeParse function can be removed, as it's not doing anything.